### PR TITLE
Switch telco-bastion to using kcli

### PR DESCRIPTION
**Pull request by johnbieren**
Switch the telco bastion workflow in ci-operator to use kcli instead of ansible-ipi-install

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

**Comment by johnbieren** [comment]
@karmab @yuvalk fyi

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh [hunk 1]
```diff
@@ -14,24 +14,25 @@ cat << EOF > ~/inventory
...
     git:
-      repo: https://github.com/openshift-kni/baremetal-deploy.git
-      dest: "~/baremetal-deploy"
+      repo: https://github.com/karamb/kcli-openshift4-baremetal.git
+      dest: "~/kcli-openshift4-baremetal"
       version: master
       force: yes
     retries: 5
+  - name: Remove old installer vm
+    shell: kcli delete vm --yes cnfdc5-installer
```
Comment (yuvalk) [inline ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:30]: why are we removing just the installed and not deleteing the plan ?
`kcli delete plan --yes upstream_ci`

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh [hunk 2]
```diff
@@ -14,24 +14,25 @@ cat << EOF > ~/inventory
...
   - name: Run deployment
-    shell: ansible-playbook -i /home/tester/inventory playbook.yml
+    shell: kcli create plan --paramfile /home/tester/kcli_parameters.yml upstream_ci --wait
```
Comment (yuvalk) [inline ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:32]: a real use case scenario, will need to tweak the parameters file, 
at least to get the right version installed

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh [hunk 3]
```diff
@@ -14,24 +14,25 @@ cat << EOF > ~/inventory
...
     args:
-      chdir: ~/baremetal-deploy/ansible-ipi-install
-  - name: Run playbook to copy kubeconfig from provisionhost vm to bastion vm
+      chdir: ~/kcli-openshift4-baremetal
+  - name: Run playbook to copy kubeconfig from installer vm to bastion vm
     shell: ansible-playbook -i /home/tester/inventory /home/tester/kubeconfig.yml
```
Comment (yuvalk) [inline ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:36]: that can be done now with kcli too
assuming installation was successful, 
kubeconfig is always under
/root/ocp/auth/kubeconfig
on the installer..

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh (hunk 1 above)
Comment (johnbieren) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:30]: That's just what I got from @karmab for rerunning. @karmab should I add or switch to `kcli delete plan` ?

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh (hunk 2 above)
Comment (johnbieren) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:32]: I believe we can pass things from the command line for this, right @karmab ? So we could remove something like the version from the kcli_parameters.yml file and pass it as an option on the cli

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh (hunk 3 above)
Comment (johnbieren) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:36]: Yeah the kubeconfig.yml playbook leverages `kcli ssh`. Are you saying you want me to copy the playbook's tasks here instead? I have to double check that nothing is internal only (there were internal things in the ansible-ipi-install scenario)

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh (hunk 1 above)
Comment (karmab) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:30]: you can do both things, but it s probably cleaner to delete the plan all together

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh [hunk 4]
```diff
@@ -14,24 +14,25 @@ cat << EOF > ~/inventory
...
     git:
-      repo: https://github.com/openshift-kni/baremetal-deploy.git
-      dest: "~/baremetal-deploy"
+      repo: https://github.com/karamb/kcli-openshift4-baremetal.git
+      dest: "~/kcli-openshift4-baremetal"
       version: master
       force: yes
     retries: 5
+  - name: Remove last run
+    shell: kcli delete plan --yes --paramfile /home/tester/kcli_parameters.yml upstream_ci
```
Comment (yuvalk) [inline ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:30]: no need to specify paramfile for kcli delete command

File: ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh (hunk 2 above)
Comment (yuvalk) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:32]: yes, that's right

Comment (yuvalk) [reply ci-operator/step-registry/telco-bastion/setup/telco-bastion-setup-commands.sh:32]: also plan name, would probably be parameterized in the futuer if/when we have more envs
but that's fine

**Pull request merged**